python/events/src/hydrotools/events/models/interception.py

```diff
@@ -0,0 +1,376 @@
+        return trunk_capacity
+
+    return (trunk_fraction * canopy_fraction *
+        (1-(1-trunk_evaporation)*evaporation_rate/rainfall_rate)*
+        (gross_rainfall-canopy_saturation))
+
+def compute_interception(
+        gross_rainfall: float,
+        rainfall_rate: float,
+        canopy_fraction: float,
+        canopy_storage: float = 1.2,
+        evaporation_rate: float = 0.3,
+        trunk_fraction: float = 0.02,
+        trunk_evaporation: float = 0.02,
+        trunk_capacity: float = 0.02
+) -> float:
+    """
+    Compute the amount of total intercepted rainfall. Typically in mm.
+
+    Parameters
+    ----------
+    gross_rainfall: float, require
+        Storm total rainfall, typically in mm. Notated $P_g$.
+    rainfall_rate: float, required
+        Mean rainfall rate in L/T, typically mm/hour. Notated 'R'.
+    canopy_fraction: float, required
+        Projected portion of canopy cover, unitless. Notated 'c'. Valid
+        range is (0.0, 1.0]
+    canopy_storage: float, optional, default 1.2
+        Minimum depth of water required to saturate the canopy. Typically 1.2
+        mm (Miralles et al, 2010). Notated $S_c$. Expected to vary little by
+        location and leaf type (± 0.4 mm, standard deviation). Measurement
+        units should correspond to rainfall_rate (i.e. if rainfall_rate is in
+        mm/hour, then canopy_storate should be in mm).
+    evaporation_rate: float, optional, default 0.3
+        Average evaporation rate. Typically 0.3 mm/hour (Miralles et al, 2010).
+        Notated $E_c$. Expected to vary little between locations
+        (± 0.1 mm/hour, standard deviation). Should be in the same measurement
+        units as rainfall_rate.
+    trunk_fraction: float, optional, default 0.02
+        Fraction of rainfall hitting the "trunks" under the canopy. Typically
+        0.02 (Miralles et al, 2010). Notated $p_d$. Unitless.
+    trunk_evaporation: float, optional, default 0.02
+        Fraction of evaporation originating from vegetation "trunks." Typically
+        0.02 (Miralles et al, 2010). Notated $\\epsilon$. Unitless.
+    trunk_capacity: float, optional, default 0.02
+        Minimum depth of water required to saturate trunks under the canopy.
+        Typically 0.02 mm (Miralles et al, 2010). Notated $S_t$. Measurement
+        units should correspond to rainfall_rate (i.e. if rainfall_rate is in
+        mm/hour, then trunk_capacity should be in mm).
+
+    Returns
+    -------
+    Depth of canopy intercepted rainfall (L) as a float.
+    """
+    # Validate
+    assert gross_rainfall > 0, "gross rainfall must be a number > 0"
+    assert rainfall_rate > 0, "rainfall rate must be a number > 0"
+    assert evaporation_rate > 0, "rainfall rate must be a number > 0"
+    assert 0.0 < canopy_fraction <= 1.0, \
+        "canopy fraction range is (0.0, 1.0]"
+    if evaporation_rate >= rainfall_rate:
+        message = ("evaporation rate >= rainfall rate, " +
```

Comment:
Unnecessary string concatenation

Reply:
Not sure what happened. I must have had my rulers set incorrectly.